### ISSUE TITLE
pythonPackages.rpy2 doesn't install correctly

### ISSUE THREAD

**Issue by bcdarwin** [NixOS/nixpkgs#16485]
With python27, the tests don't seem to run, but with python35, they fail due to missing libreadline.so.6:

$ nix-build -A python35Packages.rpy2

```
...
ImportError: Failed to import test module: rpy.tests
Traceback (most recent call last):
  File "/nix/store/w9gwivwr7ih57b4y1dfhv64xy316p5qn-python3-3.5.1/lib/python3.5/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/nix/store/w9gwivwr7ih57b4y1dfhv64xy316p5qn-python3-3.5.1/lib/python3.5/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/tmp/nix-build-python3.5-rpy2-2.5.6.drv-0/rpy2-2.5.6/rpy/tests.py", line 23, in <module>
    import rpy2.tests_rpy_classic
  File "/nix/store/za1c8jf3r7ww442wh06madf9v6ax9ap9-python3.5-rpy2-2.5.6/lib/python3.5/site-packages/rpy2/tests_rpy_classic.py", line 3, in <module>
    import rpy2.rpy_classic as rpy
  File "/nix/store/za1c8jf3r7ww442wh06madf9v6ax9ap9-python3.5-rpy2-2.5.6/lib/python3.5/site-packages/rpy2/rpy_classic.py", line 5, in <module>
    import rpy2.rinterface as ri
  File "/nix/store/za1c8jf3r7ww442wh06madf9v6ax9ap9-python3.5-rpy2-2.5.6/lib/python3.5/site-packages/rpy2/rinterface/__init__.py", line 101, in <module>
    from rpy2.rinterface._rinterface import *
ImportError: libreadline.so.6: cannot open shared object file: No such file or directory

----------------------------------------------------------------------
Ran 141 tests in 0.032s

FAILED (errors=13)
```

Running without the tests also doesn't work for other reasons (at least buildInputs that should be propagated but aren't):

```
>>> from rpy2.robjects.packages import importr
/bin/sh: 1: R: not found
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/86h3is22ql94lirb1j1pl5cvz6kfyyv7-python3.5-rpy2-2.5.6/lib/python3.5/site-packages/rpy2/robjects/__init__.py", line 15, in <module>
    import rpy2.rinterface as rinterface
  File "/nix/store/86h3is22ql94lirb1j1pl5cvz6kfyyv7-python3.5-rpy2-2.5.6/lib/python3.5/site-packages/rpy2/rinterface/__init__.py", line 48, in <module>
    "R_HOME not defined, and no R command in the PATH."
RuntimeError: R_HOME not defined, and no R command in the PATH.
```

Currently using 16.09pre83147.df89584 on Ubuntu.

I'm building some other stuff but it doesn't strictly require this package, so I'm not sure I have time to look into this at the moment.

cc @joelmo

**Comment by joelmo** [comment]
Which package are you building that pulls in this. I am not using this package right now. I think R could maybe be better as a dynamic dependency. There seems to be difference in python 2 and 3. Maybe py3 doesn't propagate readline.

**Comment by FRidh** [comment]
In 52896c8c8641c2305dd7fb9b0c551839c16d1fa3 I updated the version of `rpy2` and fixed testsuite invokation. Unfortunately I wasn't able to solve the readline issue. Propagating `readline` didn't solve the issue. In the case of Python 2.x we likely need to add `self.readline` or `modules.readline` but at this stage that didn't matter.

**Comment by teh** [comment]
The readline issue can be fixed by sticking the following line before running the tests in `checkPhase`.

```
export LD_LIBRARY_PATH=${pkgs.lib.makeLibraryPath [ pkgs.readline ]}
```

But then the tests fail because of a broken encoding. Disabling tests seems like the right thing to do. The rpy2 code doesn't work for me because patchelf shrinks out `readline` later. You can fix with something ugly like this:

```
    postFixup = let rpath = stdenv.lib.makeLibraryPath (with pkgs; [
      readline R pcre lzma bzip2 zlib icu pkgs.gcc.cc.lib
    ]) + ":${pkgs.R}/lib/R/lib/"; in ''
      find $out -name '*.so' -exec patchelf --set-rpath "${rpath}" {} \;
    '';
```

but it'd also be interesting to know why patchelf shrinks out readline

**Comment by teh** [comment]
A related issue is that I couldn't convince rpy2 to pick up any R packages like ggplot2, even with explicit environment variables or absolute paths. Not sure what's going on there.

**Comment by joelmo** [comment]
I do not know what variables you tried. I think R_HOME needs to point to some place where ggplot2 have been installed, it should be like PYTHONPATH.

**Comment by teh** [comment]
I tried a lot of variations. First, rpy2 uses the R embedding API which spawns with `--vanilla` by default (which ignores `R_LIBS`, `R_SITE_LIBS` etc, see [1]) so I removed that. Even after that the embedding library isn't picking up the libraries. I think with enough `strace`ing someone can figure this out but I don't have enough time as running R is just a hobby.

[1]

> The command-line option --vanilla implies --no-site-file, --no-init-file, --no-environ and (except for R CMD) --no-restore

from https://stat.ethz.ch/R-manual/R-devel/library/base/html/Startup.html
